lib/server.js

```diff
@@ -164,6 +184,59 @@ module.exports = function createServer(config, log) {
   )
 
   api.post(
+    '/checkAuthenticated',
+    function (req, res, next) {
+      var action = req.body.action
+      var ip = req.body.ip
+      var uid = req.body.uid
+
+      if(!action || !ip || !uid){
+        var err = {code: 'MissingParameters', message: 'action, ip and uid are all required'}
+        log.error({op:'request.check', action: action, ip: ip, uid: uid, err: err})
+        res.send(400, err)
+        return next()
+      }
+
+      fetchUidRecord(uid)
+        .spread(
+          function (uidRecord) {
+
+            var retryAfter = uidRecord.addCount(action, uid)
+
+            return setUidRecord(uid, uidRecord)
+              .then(
+                function () {
+
+                  return {
+                    block: retryAfter > 0,
+                    retryAfter: retryAfter,
+                    uid: uid,
+                    uidRecord: uidRecord
+                  }
+                }
+              )
+          }
+        )
+        .then(
+          function (result) {
+            log.info({ op: 'request.check', block: result.block })
```

Comment:
@vladikoff need to check if this is the best metrics we can get and if it is enough.